lib/pbench/server/unpack_tarballs.py

```diff
@@ -0,0 +1,153 @@
+from logging import Logger
+import os
+from pathlib import Path
+import tempfile
+from typing import NamedTuple
+
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.report import Report
+from pbench.server.utils import get_tarball_md5
+
+
+class Results(NamedTuple):
+    total: int
+    success: int
+
+
+class UnpackTarballs:
+    LINKSRC = "TO-UNPACK"
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        """Unpacks Tarball in the INCOMING Directory
```

Comment:
This is really more of a class docstring; the `__init__` method does no such thing. 😁